src/plugins/external_tool_support/src/blast/BlastCommonTask.h

```diff
@@ -62,16 +63,26 @@ class BlastCommonTask : public ExternalToolSupportTask {
     ExternalToolRunTask *blastTask = nullptr;
     U2SequenceObject *sequenceObject = nullptr;
     Document *tmpDoc = nullptr;
-    QList<SharedAnnotationData> result;
-    U2PseudoCircularization *circularization = nullptr;
+
+    /** Results by sequence index in settings.querySequences list. */
+    QHash<int, QList<SharedAnnotationData>> resultsPerQuerySequence;
+
+    /** Pre-processed and 'circularized' query sequences. */
+    QList<QByteArray> querySequences;
 
     void parseTabularResult();
     void parseTabularLine(const QByteArray &line);
 
     void parseXMLResult();
-    void parseXMLHit(const QDomNode &xml);
-    void parseXMLHsp(const QDomNode &xml, const QString &id, const QString &def, const QString &accession);
+    void parseXMLHit(const QDomNode &xml, int querySequenceIndex);
+    void parseXMLHsp(const QDomNode &xml, int querySequenceIndex, const QString &id, const QString &def, const QString &accession);
     QString getAcceptableTempDir() const;
+
+    /**
```

Comment:
5 spaces indent

Reply:
All is good with indentation here. Sorry, do not get it.